endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java

```diff
@@ -422,8 +428,8 @@ private DirectoryList generateDirectory(Map<ApiKey, RestDescription> discoveryDo
           .setDiscoveryLink("." + relativePath)
           .setDiscoveryRestUrl(context.getApiRoot() + "/discovery/v1" + relativePath)
           .setIcons(new Icons()
-                .setX16("https://www.gstatic.com/images/branding/product/1x/googleg_16dp.png")
-                .setX32("https://www.gstatic.com/images/branding/product/1x/googleg_32dp.png"))
+              .setX16(doc.getIcons().getX16())
+              .setX32(doc.getIcons().getX32()))
```

Comment:
A fallback to the (new?) google default icons would make this feature backward compatible.

Reply:
It is backward compatible: the default icons are set in REST_SKELETON, which is cloned to generate RestDescription instances in DicoveryGenerator (the previous code was actually useless ...).

Reply:
Ok cool, I missed it.